analyzeDelphes.py

```diff
@@ -50,295 +82,267 @@ def isBeamRemnant(p):
+        photons  =selector(event.Photon,                   'x.PT>%f and abs(x.Eta)<%f'%(pTMin[22],etaMax[22]))
+        jets     =selector(event.__getattr__(jetType),     'x.PT>%f and abs(x.Eta)<%f'%(pTMin[21],etaMax[21]))
+
+        #electrons=selector(event.Electron,'x.PT>5 and abs(x.Eta)<2')
+        #muons=selector(event.Muon,'x.PT>5 and abs(x.Eta)<2')
+
+        #fMuons=selector(event.Muon,'abs(x.Eta)>2')
+        #fMuons+=selector(event.Electron,'abs(x.Eta)>2 and x.Particle.GetObject().PID==11 and x.Particle.GetObject().Status==1 and x.Particle.GetObject().M1<5')  #this is a hack - not a typo
+       
+        #sort object lists
+        for collection in [electrons, muons, photons, jets]: collection.sort(key=operator.attrgetter('PT'),reverse=True)
+
+        h_reco[11]['mult'].Fill(len(electrons))
+        for i in range(len(electrons)):
+            P4_f+=electrons[i].P4()
+            h_reco[11]['pT']['I'].Fill(electrons[i].PT)
+            h_reco[11]['p']['I'].Fill(electrons[i].P4().P())
+            h_reco[11]['eta']['I'].Fill(electrons[i].Eta)
+
+            if i<4:
+                h_reco[11]['pT'][i].Fill(electrons[i].PT)
+                h_reco[11]['p'][i].Fill(electrons[i].P4().P())
+                h_reco[11]['eta'][i].Fill(electrons[i].Eta)
+
+        h_reco[13]['mult'].Fill(len(muons))
+        for i in range(len(muons)):
+            P4_f+=muons[i].P4()
+            h_reco[13]['pT']['I'].Fill(muons[i].PT)
+            h_reco[13]['p']['I'].Fill(muons[i].P4().P())
+            h_reco[13]['eta']['I'].Fill(muons[i].Eta)            
+            if i<4:
+                h_reco[13]['pT'][i].Fill(muons[i].PT)
+                h_reco[13]['p'][i].Fill(muons[i].P4().P())
+                h_reco[13]['eta'][i].Fill(muons[i].Eta)
+
+        h_reco[22]['mult'].Fill(len(photons))
+        for i in range(len(photons)):
+            P4_f+=photons[i].P4()
+            h_reco[22]['pT']['I'].Fill(photons[i].PT)
+            h_reco[22]['p']['I'].Fill(photons[i].P4().P())
+            h_reco[22]['eta']['I'].Fill(photons[i].Eta)
+            if i<4:
+                h_reco[22]['pT'][i].Fill(photons[i].PT)
+                h_reco[22]['p'][i].Fill(photons[i].P4().P())
+                h_reco[22]['eta'][i].Fill(photons[i].Eta)
+
+        h_reco[11]['mult'].Fill(len(jets))
+        for i in range(len(jets)):
+            P4_f+=jets[i].P4()
+            h_reco[21]['pT']['I'].Fill(jets[i].PT)
+            h_reco[21]['p']['I'].Fill(jets[i].P4().P())
+            h_reco[21]['eta']['I'].Fill(jets[i].Eta)
+            if i<4:
+                h_reco[21]['pT'][i].Fill(jets[i].PT)
+                h_reco[21]['p'][i].Fill(jets[i].P4().P())
+                h_reco[21]['eta'][i].Fill(jets[i].Eta)
+        
+        R_missingMass.Fill((P4_f-P4_i).M())
+
+        #-------------------------------------------------------------------------------------------------
+        leptons=electrons+muons
+
```

Comment:
This is where we want to add some code.  First, lets define a new kind of object, a "W jet."  W jets should have mass in the 50-100 GeV range (just looking at your leading jet mass plot).

Then, you should check and make sure you found two beam remnant muons and 2 W jets.  If not, you can just skip to the next event with a `continue` statement.  This is the "object selection" we want to apply.